src/api/api.ts

```diff
@@ -0,0 +1,49 @@
+import { taskEither as TE } from "fp-ts";
+import { identity } from "fp-ts/lib/function";
+import { BusinessesApi, Business } from "../models/domain";
+import { pipe } from "fp-ts/lib/pipeable";
+import { Type, TypeOf } from "io-ts";
+
+function getApiCall<T>(
+  relativeUrl: string,
+  params: object,
+  decoder: Type<T>
+): TE.TaskEither<unknown, TypeOf<typeof decoder>> {
+  const query = new URLSearchParams(params as any).toString();
+  return pipe(
+    TE.tryCatch(
+      () =>
+        fetch(
+          `${process.env.REACT_APP_YELP_BASE_API_URL}${relativeUrl}${
+            query ? `?${query}` : ""
+          }`,
+          {
+            headers: {
+              Authorization: `Bearer ${process.env.REACT_APP_YELP_API_KEY}`,
+            },
+          }
+        ),
+      identity
+    ),
+    TE.fold(
+      (error) => TE.left(error),
+      (res) => TE.tryCatch(() => res.json(), identity)
+    ),
+    TE.fold(
+      (error) => TE.left(error),
+      (res) => TE.fromEither(decoder.decode(res))
+    )
+  );
+}
+
+export function getBusinessesList(
+  location: string
+): TE.TaskEither<unknown, Business[]> {
+  return pipe(
+    getApiCall("businesses/search", { location }, BusinessesApi),
+    TE.fold(
+      (error) => TE.left(error),
+      (res) => TE.right(res.businesses)
+    )
```

Comment:
similar to my comment above, `TaskEither` (as any other data type for which a `Functor` instance is defined) defines `map`:
```suggestion
    TE.map(res => res.businesses)
```

Reply:
Also notice that `fold` de-structures the `TaskEither`: both for `left` and `right` you define functions which can return any type and that's the returned type of the `fold`.
Indeed since you still wanted a `TaskEither` you had to return a `TE.left` and a `TE.right`.

`map`,`mapLeft`,`bimap` let you apply transformations to `right`, `left` or both, preserving the structure.